pennylane/data/attributes/pauli.py

```diff
@@ -0,0 +1,44 @@
+import json
+from typing import Tuple, Type
+
+from pennylane.data.base.attribute import AttributeType
+from pennylane.data.base.typing_util import HDF5Array, HDF5Group
+from pennylane.pauli import PauliSentence, PauliWord
+
+
+class DatasetPauliWord(AttributeType[HDF5Array, PauliWord, PauliWord]):
+    type_id = "pauli_word"
+
+    @classmethod
+    def consumes_types(cls) -> Tuple[Type[PauliWord]]:
+        return (PauliWord,)
+
+    def hdf5_to_value(self, bind: HDF5Array) -> PauliWord:
+        return PauliWord(json.loads(bind.asstr()[()]))
+
+    def value_to_hdf5(self, bind_parent: HDF5Group, key: str, value: PauliWord) -> HDF5Array:
+        bind_parent[key] = json.dumps(dict(value.items()))
+
+        return bind_parent[key]
+
+
+class DatasetPauliSentence(AttributeType[HDF5Group, PauliSentence, PauliSentence]):
+    type_id = "pauli_sentence"
+
+    @classmethod
+    def consumes_types(cls) -> Tuple[Type[PauliSentence]]:
+        return (PauliSentence,)
+
+    def hdf5_to_value(self, bind: HDF5Group) -> PauliSentence:
+        return PauliSentence(
+            (PauliWord(json.loads(word)), coeff)
+            for word, coeff in zip(bind["words"], bind["coeffs"])
+        )
+
+    def value_to_hdf5(self, bind_parent: HDF5Group, key: str, value: PauliWord) -> HDF5Group:
```

Comment:
Are there any performance implications with respect to serializing a data structure using
```python
{
    "coeffs": [4, 5, 6],
    "words":  ['{"1": "X"}', '{"2": "Y"}', '{"3": "Z"}']
}
```
as opposed to
```python
[
    {
        "coeff": 4,
        "word": '{"1": "X"}'
    },
    {
        "coeff": 5,
        "word": '{"2": "Y"}'
    },
    {
        "coeff": 6,
        "word": '{"3": "Z"}'
    }
]
```

Reply:
The first might be slightly faster, but It's not so much a performance issue, it's that we don't want to use JSON to serialize the coefficients.